pkg/s3wrapper/client.go

```diff
@@ -341,3 +346,70 @@ func (c *S3Client) handleObject(ctx context.Context, log logrus.FieldLogger, obj
 		}
 	}
 }
+
+func (c *S3Client) ListObjectsByPrefix(ctx context.Context, prefix string) ([]string, error) {
+	log := logutil.FromContext(ctx, c.log)
+	var objects []string
+	log.Info("Listing objects by with prefix %s", prefix)
+	resp, err := c.client.ListObjects(&s3.ListObjectsInput{
+		Bucket: aws.String(c.cfg.S3Bucket),
+		Prefix: aws.String(prefix),
+	})
+	if err != nil {
+		err = errors.Wrapf(err, "Error listing objects for prefix %s", prefix)
+		log.Error(err)
+		return nil, err
+	}
+	for _, key := range resp.Contents {
+		objects = append(objects, *key.Key)
+	}
+	return objects, nil
+}
+
+func (c *S3Client) DownloadListOfFiles(ctx context.Context, files []string) (io.ReadCloser, int64, error) {
+	log := logutil.FromContext(ctx, c.log)
+	// Create a new stream.
+	w, err := stream.New("downloads")
+	if err != nil {
+		err = errors.Wrapf(err, "Failed tp create stream for files %s", files)
+		log.Error(err)
+		return nil, 0, err
+	}
+	defer w.Close()
+
+	// Create a new tar archive.
+	tarWriter := tar.NewWriter(w)
+	var rdr io.ReadCloser
+	var objectSize int64
+
+	// Create tar headers from s3 files
+	for _, file := range files {
+
+		// Read file from S3, log any errors
+		rdr, objectSize, err = c.Download(ctx, file)
+		if err != nil {
+			log.WithError(err).Warnf("Failed to open reader for %s, skipping it", file)
+			continue
+		}
+
+		header := tar.Header{
+			Name: file,
+			Size: objectSize,
+		}
+
+		_ = tarWriter.WriteHeader(&header)
+		_, _ = io.Copy(tarWriter, rdr)
```

Comment:
Does it copy the content into the object? if it is then it may take a lot of memory

Reply:
nope, it doesn't. it only copies reader itself. All this code is taking almost 0 memory

Reply:
that was the poing